modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/DistributionZoneCausalityDataNodesTest.java

```diff
@@ -1013,62 +996,128 @@ void simpleTopologyChanges() throws Exception {
+        blockDataNodesUpdatesInMetaStorage(latch);
 
         Set<LogicalNode> topology1 = Set.of(NODE_0, NODE_1, NODE_2);
-        Set<LogicalNode> topology2 = Set.of(NODE_0, NODE_2);
 
         // Change logical topology. NODE_2 is added.
-        long topologyRevision1 = putNodeInLogicalTopologyAndGetRevision(NODE_2, topology1);
-        waitForCondition(() -> metaStorageManager.appliedRevision() >= topologyRevision1, TIMEOUT);
+        long topologyRevision = putNodeInLogicalTopologyAndGetRevision(NODE_2, topology1);
+        waitForCondition(() -> metaStorageManager.appliedRevision() >= topologyRevision, TIMEOUT);
 
-        // Change logical topology. NODE_1 is removed.
-        long topologyRevision2 = removeNodeInLogicalTopologyAndGetRevision(Set.of(NODE_1), topology2);
-        waitForCondition(() -> metaStorageManager.appliedRevision() >= topologyRevision2, TIMEOUT);
+        assertEquals(topologyRevision, topologyUpdateRevision.get());
 
-        Set<String> expectedDefaultZoneDataNodes = Set.of(NODE_0.name(), NODE_2.name());
-        Set<String> expectedZone1DataNodes = Set.of(NODE_0.name(), NODE_1.name(), NODE_2.name());
-        Set<String> expectedZone2DataNodes = Set.of(NODE_0.name());
-        Set<String> expectedZone3DataNodes = Set.of(NODE_0.name(), NODE_1.name());
+        latch.await();
+
+        // Check that data nodes values are idempotented.
+        // Data nodes from the meta storage manager only used to calculate current data nodes because the meta storage are updated and
+        // topology augmentation maps were cleared.
+        checkDataNodes(
+                topologyUpdateRevision.get(),
+                expectedDefaultZoneDataNodes,
+                expectedZone1DataNodes,
+                expectedZone2DataNodes,
+                expectedZone3DataNodes
+        );
 
-        // Check that data nodes values are changed according to scale up and down timers.
-        // Data nodes from the meta storage manager and topology augmentation maps are used to calculate current data nodes.
         checkDataNodes(
-                topologyRevision2,
+                metaStorageManager.appliedRevision(),
                 expectedDefaultZoneDataNodes,
                 expectedZone1DataNodes,
                 expectedZone2DataNodes,
                 expectedZone3DataNodes
         );
+    }
+
+    /**
+     * Check that data nodes calculation is idempotented.
+     * The current data nodes value which was calculated by data nodes from the meta storage manager and topology augmentation maps equals
+     * to the data nodes value which were calculated by data nodes from the meta storage manager after topology augmentation maps
+     * were cleared.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    void checkDataNodesRepeatedOnNodeRemoved() throws Exception {
+        prepareZonesWithTwoDataNodes();
+
+        prepareZonesTimerValuesToTest();
```

Comment:
Lets add zone with non-immediate timers, but not infinite (1 second)

Reply:
Fixed. Added this zone.